src/Orleans.Core/Core/ClusterClient.cs

```diff
@@ -72,13 +94,6 @@ private IInternalGrainFactory InternalGrainFactory
         private bool IsDisposing => this.state == LifecycleState.Disposed ||
                                     this.state == LifecycleState.Disposing;
 
-        /// <inheritdoc />
-        public IEnumerable<IStreamProvider> GetStreamProviders()
-        {
```

Comment:
Providers in general are now just natural extensions to Orleans.  No special support is needed for the behaviors,  The legacy provider support introduced in this PR is there only to allow externally developed providers to continue to work.

These functions no longer fit how Orleans works.  We can continue to support them in the legacy packages via extension functions, but I'm not convinced that is necessary or good.  Further, in these cases, I don't think these calls are commonly used, so we'd get little by maintaining them.

Reply:
I'd note that their is a meaningful difference between providers and logging.  Most systems need logging and those that use it will use the related calls a lot.  Providers have far less application layer surface and their api's will likely be used far less often.  This means the impact of breaking change is dramatically less with providers than logging.